spring-boot-project/spring-boot/src/main/java/org/springframework/boot/webservices/client/WebServiceTemplateBuilder.java

```diff
@@ -0,0 +1,1106 @@
+				this.webServiceMessageSenderSuppliers,
+				this.webServiceMessageSenderCustomizers, this.marshaller,
+				this.unmarshaller, this.destinationProvider, this.transformerFactoryClass,
+				this.messageFactory, this.detectWebServiceMessageSender);
+	}
+
+	/**
+	 * Sets the {@code Suppliers} of {@link WebServiceMessageSender} that should be called
+	 * each time when {@link #configure(WebServiceTemplate)} method is called.
+	 *
+	 * <b>Note!</b> Override existing WebServiceMessageSender {@code suppliers}
+	 * @param webServiceMessageSenderSuppliers Suppliers for the messageSenders
+	 * @return a new builder instance.
+	 * @see WebServiceTemplate#setMessageSenders(WebServiceMessageSender[])
+	 */
+
+	public WebServiceTemplateBuilder setWebServiceMessageSenders(
+			Collection<? extends Supplier<? extends WebServiceMessageSender>> webServiceMessageSenderSuppliers) {
+		Assert.notNull(webServiceMessageSenderSuppliers,
+				"webServiceMessageSenderSuppliers must not be null");
+		return new WebServiceTemplateBuilder(this.interceptors, this.internalCustomizers,
+				this.customizers,
+				append(Collections
+						.<Supplier<? extends WebServiceMessageSender>>emptySet(),
+						webServiceMessageSenderSuppliers),
+				this.webServiceMessageSenderCustomizers, this.marshaller,
+				this.unmarshaller, this.destinationProvider, this.transformerFactoryClass,
+				this.messageFactory, this.detectWebServiceMessageSender);
+	}
+
+	/**
+	 * Add additional {@code Suppliers} of {@link WebServiceMessageSender} that should be
+	 * called each time when {@link #configure(WebServiceTemplate)} method is called.
+	 * @param webServiceMessageSenderSuppliers Suppliers for the messageSenders
+	 * @return a new builder instance.
+	 * @see WebServiceTemplate#setMessageSenders(WebServiceMessageSender[])
+	 */
+	public WebServiceTemplateBuilder addWebServiceMessageSenders(
+			Collection<? extends Supplier<? extends WebServiceMessageSender>> webServiceMessageSenderSuppliers) {
+		Assert.notNull(webServiceMessageSenderSuppliers,
+				"webServiceMessageSenderSuppliers must not be null");
+		return new WebServiceTemplateBuilder(this.interceptors, this.internalCustomizers,
+				this.customizers,
+				append(this.webServiceMessageSenderSuppliers,
+						webServiceMessageSenderSuppliers),
+				this.webServiceMessageSenderCustomizers, this.marshaller,
+				this.unmarshaller, this.destinationProvider, this.transformerFactoryClass,
+				this.messageFactory, this.detectWebServiceMessageSender);
+	}
+
+	/**
+	 * Set {@link WebServiceTemplate#setCheckConnectionForFault(boolean)
+	 * setCheckConnectionForFault} on the underlying.
+	 * @param checkConnectionForFault Specify whether checkConnectionForFault should be
+	 * enabled or not.
+	 * @return a new builder instance.
+	 **/
+	public WebServiceTemplateBuilder setCheckConnectionForFault(
+			boolean checkConnectionForFault) {
+		return new WebServiceTemplateBuilder(this.interceptors,
+				append(this.internalCustomizers,
```

Comment:
I don't like this pattern. If you call the setter several times, you'll append this customizer and rely on the fact the last one called will set the expected value. I guess you've done this that way to avoid adding too much parameters to the builder? I think we need to find a different option for this.

Reply:
it could be handled by `Boolean`  type, but you still need a custom customizer for `FaultMessageResolver`. 

Yes, I did this to avoid adding too many parameters.

Reply:
In the meantime I've seen we're doing that with `RestTemplateBuilder` and I see you're using a separate collection. Retrospectively, it's not that bad at all.